tfexec/state_rm.go

```diff
@@ -0,0 +1,104 @@
+package tfexec
+
+import (
+	"context"
+	"os/exec"
+	"strconv"
+)
+
+type stateRmConfig struct {
+	backup      string
+	backupOut   string
+	dryRun      bool
+	lock        bool
+	lockTimeout string
+	state       string
+	stateOut    string
+}
+
+var defaultStateRmOptions = stateRmConfig{
+	lock:        true,
+	lockTimeout: "0s",
+}
+
+// StateRmCmdOption represents options used in the Refresh method.
+type StateRmCmdOption interface {
+	configureStateRm(*stateRmConfig)
+}
+
+func (opt *BackupOption) configureStateRm(conf *stateRmConfig) {
+	conf.backup = opt.path
+}
+
+func (opt *BackupOutOption) configureStateRm(conf *stateRmConfig) {
+	conf.backupOut = opt.path
+}
+
+func (opt *DryRunOption) configureStateRm(conf *stateRmConfig) {
+	conf.dryRun = opt.dryRun
+}
+
+func (opt *LockOption) configureStateRm(conf *stateRmConfig) {
+	conf.lock = opt.lock
+}
+
+func (opt *LockTimeoutOption) configureStateRm(conf *stateRmConfig) {
+	conf.lockTimeout = opt.timeout
+}
+
+func (opt *StateOption) configureStateRm(conf *stateRmConfig) {
+	conf.state = opt.path
+}
+
+func (opt *StateOutOption) configureStateRm(conf *stateRmConfig) {
+	conf.stateOut = opt.path
+}
+
+// StateRm represents the terraform state rm subcommand.
+func (tf *Terraform) StateRm(ctx context.Context, source string, opts ...StateRmCmdOption) error {
```

Comment:
`address` would be a clearer name for this parameter than `source` here, I think, corresponding to the cmd help text:

```
$ terraform -help state rm
Usage: terraform state rm [options] ADDRESS...
```

Reply:
Yep, I agree will fix 👍🏻